test/test_linalg.py

```diff
@@ -7485,6 +7485,19 @@ def run_test(A_dims, b_dims, pivot=True):
         run_test((4, 4), (2, 1, 3, 4, 2))  # broadcasting A
         run_test((1, 3, 1, 4, 4), (2, 1, 3, 4, 5))  # broadcasting A & b
 
+    @onlyCUDA
+    @skipCUDAIfNoMagma
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    # this tests https://github.com/pytorch/pytorch/issues/36921
+    def test_lu_solve_large_matrices(self, device, dtype):
+        def run_test(A_dims, b_dims):
```

Comment:
I think the name is a bit confusing. `run_tests` gets `A_dims = (1, 1)`, `b_dims = (1, 1, 1025)`. So, effectively, matrices are small!

Reply:
Fair, how's `test_lu_solve_large_b_second_dim`?

Reply:
The referenced issue actually considers batched large matrices. Why wouldn't we test all the cases? I.e., large batch small size, small batch large size?

Reply:
The only remaining issue in the current master branch is with b tensors with a last dimension > 1024 due to the linked bug in MAGMA. IvanYashchuk asked for a test which fails on the master branch but passes after this PR. These other tests would pass in the master branch as well, but I can add them anyway if you'd like.